File: src/sampleCode/factorial.js
function factorial(n) {
  // If the number is negative
  if (n < 0) {
    return "Negative numbers don't have factorials";
  }
  // If the number is 0 or 1
  else if (n === 0 || n === 1) {
    return 1;
  }
  // For numbers greater than 1
  else {
    let fact = 1;
    // Calculate factorial using a loop
    for (let i = 1; i <= n; i++) {
      fact *= i;
    }
    return fact;
  }
}

// Test the program
const number = 5;
const result = factorial(number);
console.log(`factorial ${number} is equal to: ${result}`);

File: SimpleCode/factorial.py
def factorial(n):
    # اگر عدد منفی باشد
    if n < 0:
        return "عدد منفی فاکتوریل ندارد"
    # اگر عدد 0 یا 1 باشد
    elif n == 0 or n == 1:
        return 1
    # برای اعداد بزرگتر از 1
    else:
        fact = 1
        # محاسبه فاکتوریل با حلقه
        for i in range(1, n + 1):
            fact = fact * i
        return fact

# تست برنامه
number = 5
result = factorial(number)
print(f"factorial {number} is equal to: {result}")